transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -65,7 +65,14 @@
         for (int i = 0; i < iovArrays.length; i++) {
             iovArrays[i] = new IovArray();
         }
-        ringBuffer = Native.createRingBuffer();
+        ringBuffer = Native.createRingBuffer(new Runnable() {
+            @Override
+            public void run() {
+                // Once we submitted its safe to clear the IovArrays and so be able to re-use these.
+                clearUsedIovArrays();
+            }
+        });
```

Comment:
I increasingly want the `IovArray[8]` to be encapsulated in its own component, which we could then ask for a clear-callback. Then we could also test it independently.

Reply:
I can do this... But lets do this as a followup..